ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java

```diff
@@ -943,9 +903,24 @@ private boolean isStringInUrl(String toFind, int timeoutSeconds) {
         return result;
     }
 
-    @When("^set 3scale discovery variable to \"([^\"]*)\"")
+    @When("^enable 3scale discovery with url \\\"([^\\\"]*)\\\"")
     public void set3scaleEnvVar(String value) {
-        OpenShiftUtils.updateEnvVarInDeploymentConfig("syndesis-server", "CONTROLLERS_EXPOSE_VIA3SCALE", value);
+
+        Map cr = SyndesisTemplate.getDeployedCr();
+        Map features = (Map<String, Object>) ((Map<String, Object>) ((Map<String, Object>) ((Map<String, Object>) cr.get("spec")).get("components"))
+            .get("server")).get("features");
+
+        if (!"".equals(value)) {
+            features.put("managementUrlFor3scale", value);
+        } else {
+            features.remove("managementUrlFor3scale");
+        }
```

Comment:
Can't say I'm super excited about this - the step is is called `enable 3scale discovery` and here we have a magic value that will disable it. Maybe add a separate step for disabling?

Reply:
Indeed, it is not the best solution, refactored it so there are two public methods for enabling and disabling